packages/camera/camera_android/android/src/main/java/io/flutter/plugins/camera/Camera.java

```diff
@@ -256,10 +278,19 @@ private void prepareMediaRecorder(String outputFilePath) throws IOException {
 
     // TODO(camsim99): Revert changes that allow legacy code to be used when recordingProfile is null
     // once this has largely been fixed on the Android side. https://github.com/flutter/flutter/issues/119668
-    if (SdkCapabilityChecker.supportsEncoderProfiles() && getRecordingProfile() != null) {
-      mediaRecorderBuilder = new MediaRecorderBuilder(getRecordingProfile(), outputFilePath);
+    EncoderProfiles recordingProfile = getRecordingProfile();
```

Comment:
Why have you moved a call outside of the check that the SDK supports the call?

Reply:
moved back